segments/custom/README.md

```diff
@@ -0,0 +1,65 @@
+# Custom_*
+
+## Installation
+
+This is not an ordinary segment that can just be installed. This is the
+documentation for the mechanism that is used to create own custom segments
+that can display output of arbitrary commands. In this documentation we
+will create a custom segment `custom_wifi_signal` to display the current
+signal strenght. You can replace this code with your own if you want to
+achieve something different. You have to preserve the prefixes though.
+This means `P9K_CUSTOM_*` for variables and `custom_*` when putting the
+custom segemnt into your `P9K_LEFT_PROMPT_ELEMENTS` or
+`P9K_RIGHT_PROMPT_ELEMENTS`.
+
+## Configuration
+
+### Command as Segment
+
+```zsh
+P9K_CUSTOM_WIFI_SIGNAL="echo signal: \$(nmcli device wifi | grep yes | awk '{print \$8}')"
+P9K_CUSTOM_WIFI_SIGNAL_BACKGROUND="blue"
+P9K_CUSTOM_WIFI_SIGNAL_FOREGROUND="yellow"
+P9K_LEFT_PROMPT_ELEMENTS=(context time battery dir vcs virtualenv custom_wifi_signal)
+```
+
+### Function as Segment
+If you prefer, you can also define the function in your `.zshrc` rather than
+putting it in-line with the variable export, as shown above. Just don't forget
+to invoke your function from your segment! Example code that achieves the same
+result as the above:
```

Comment:
as the above, with the added benefit of different foreground colors depending on the signal strength: